src/main/webapp/app/exam/participate/exam-participation.component.ts

```diff
@@ -507,4 +514,49 @@ export class ExamParticipationComponent implements OnInit, OnDestroy, ComponentC
         this.synchronizationAlert$.next();
         console.error(error);
     }
+
+    private createProgrammingExerciseSubmission(exerciseId: number, participationId: number): Subscription {
```

Comment:
Is there a reason why you make the subscription for the submission in `exam-participation.component.ts`? Wouldn't it make more sense to do it in the `programming-exam-submission.component.ts`?

Reply:
Because the programming exam submission gets destroyed when we navigate to the end page and summary page. When the student triggers a build right before the end, it is more probable, that the latest submission will be displayed in the summary page.